### ISSUE TITLE
Not able to connect to existing implicit datasource (latest 3.3.5 version)

### ISSUE THREAD

**Issue by verscev80** [TabularEditor/TabularEditor3#647]
I installed the latest version of Tabular Editor, but it is no longer possible to connect to my datasource.
First I receive the following error:

**A connection was successfully established with the server, but then an error occurred during the login process. (provider: SSL Provider, error 0 - The target principal name is incorrect**

After checking the connection details, I noticed that the 'encrypt connection' flag was set to true, and this should not be the case. After deselecting and clicking OK, Tabular Editor goes in an unhandled exception. (see attachment)

![Snag_adc802a](https://user-images.githubusercontent.com/102680661/188836665-dc3c3f30-5c33-43ef-8cbf-0b27b6e6a58d.png)
![Snag_adca277](https://user-images.githubusercontent.com/102680661/188836695-d41e8793-9f99-4640-9adf-d989565ffd84.png)

It is an existing connection a was working before.

**Comment by otykier** [comment]
Hi @verscev80 

Does this happen when attempting to import new tables using the Import Table Wizard, or when performing a schema update?
Could you share the full error report content? Thanks!

**Comment by verscev80** [comment]
When I update schema, it will give me the 'target principle' error, but nothing happens when I deselect the 'encrypt connection' flag. If I would like to add new tables through the wizard, it will give also the 'target principle' error and the unexpected error once I deselect the 'encrypt connection' flag. 
The only comment I see in the error report is the one above. Can you tell me where to find the full error report?
Thank you.

**Comment by otykier** [comment]
Thanks. Upon closer look there is no more details available in the error report for this error. Could you share what your connection string / implicit data source looks like? (Please blank out any sensitive information)

**Comment by verscev80** [comment]
We have direct connections in the partitions, like this:

![Snag_b17f617](https://user-images.githubusercontent.com/102680661/188851286-cd3a73a2-53b3-4d3c-9a34-098b5acb50c5.png)

**Comment by otykier** [comment]
Thanks again. What type of SQL Server is this? (On-prem, Azure SQL Database, Synapse SQL Pool, something else?)

**Comment by verscev80** [comment]
It is an on premise database.
Strange this is that is have been working like this since over a year.
As a test, I created a new model to import one table using the same database, and even then I receive the same issues.

**Comment by otykier** [comment]
@verscev80 just to confirm, are you using Windows Authentication when signing in to the SQL Server? Also, have you changed your Windows password recently? If so, could you try to sign out of your Windows session and then sign in again? Thanks.

**Comment by verscev80** [comment]
Yes, I am using Windows Authentication. And my credentials have not changed,

**Comment by otykier** [comment]
Thanks for the confirmation. May I ask you to check one last thing? Next to your Model.bim or Database.json file, you should see a .tmuo file. Could you open this file in a text editor and check that it has the "Trust Server Certificate" option set to "True" like below:

```json
{
  "TableImportSettings": {
    "SQL/localhost;AdventureWorksDW2019": {
      "ServerType": "Sql",
      "Options": {
        "Trust Server Certificate": "True"
      },
      "NeedsNewCredentials": false,
      "Server": "localhost",
      "Database": "AdventureWorksDW2019",
      "Authentication": "WindowsIntegrated",
      "Encrypt": true
    }
  }
}
```

If not, could you try to add that option, and see if that solves the problem?

**Comment by otykier** [comment]
Fixed in [3.3.6](https://github.com/TabularEditor/TabularEditor3/releases/tag/3.3.6), thanks for your patience!
